test/jdk/java/rmi/server/UnicastRemoteObject/serialFilter/FilterUROTest.java

```diff
@@ -70,7 +71,7 @@ static Object[][] bindObjects() {
     @Test(dataProvider = "bindData")
     public void useExportObject(String name, Object obj, int expectedFilterCount) throws RemoteException {
         try {
-            RemoteImpl impl = RemoteImpl.create();
+            impl = RemoteImpl.create();
```

Comment:
I'd prefer to use  ReachabilityFence at the end of the try block to keep the nested scoping.
```Reference.reachabilityFence(impl);```

Reply:
I just tested it and it really be OK, I'll submit a new commit.

Reply:
yes, this mitigates the issue within the test,  and alls good.
BUT it still leave an open question as to why the behaviour of the test is different for the -Xcomp mode and the interpretative mode?
I think it would be reasonable to expect both modes to behave the same. As such, that the compile mode should generate a fence or whatever to guarantee that the impl remain a strong reference until it goes out of scope at the end of the try block ?
Thus is the case in non -Xcomp mode, but in -Xcomp the status of the impl reference is accelerate to to being unreachable and a candidate for garbage collection ?
Is it not a hotspot compiler issue or the component area responsible for -Xcomp ?

Reply:
Would this behaviour be covered by the JLS and JVM specs ?

Reply:
The JLS and JVMS specs do not constrain the implementation of GC.
The reachabilityFence() was introduced in JDK 9 by https://bugs.openjdk.org/browse/JDK-8133348.